src/server/engine_shard.cc

```diff
@@ -661,13 +661,17 @@ void EngineShard::Heartbeat() {
 
   CacheStats();
 
+  // TODO: iterate over all namespaces
+  DbSlice& db_slice = namespaces.GetDefaultNamespace().GetDbSlice(shard_id());
+  // Skip heartbeat if we are serializing a big value
+  if (db_slice.HasBlockingCounterMutating()) {
```

Comment:
we should add some prints here so we will be able to know if hearbeat is not running for long time maybe because we are running lots of big values serialization or we have a bug somewere causing this skip

Reply:
+100 on this one. I thought the same tbh